spec/acceptance/rules_spec.rb

```diff
@@ -241,7 +240,7 @@ class { '::firewall': }
     it 'applies cleanly' do
       # Run it twice and test for idempotency
       apply_manifest(pp2, catch_failures: true)
-      apply_manifest(pp2, catch_changes: do_catch_changes)
+      apply_manifest(pp2, catch_changes: true) unless fact('operatingsystem') == 'SLES' && fact('operatingsystemmajrelease') == '11'
```

Comment:
You seem to skip the idempotent checks, is there a reason for this?

Reply:
on sles 11 sp4 State has been deprecated and given a warning to use ctstate instead. It seems to do this for you automatically so everytime the manifest runs it applies as state, but then the firewall changes this to ctstate

Reply:
I feel like this needs to be a code change instead? If we are passing through the wrong flag we are always going to get those change 'failures', whereas I think the commands being ran need updated to use the -ctstate flag instead of -state?

Reply:
Have been looking at using an if to swap between the two for sless 11.4 but replacing all mentions of state with ctstate might be good as well. They are simply different packages you can choose between. Ctstate just has more features and complexity.